src/EbmlMaster.cpp

```diff
@@ -117,7 +117,7 @@ std::uint64_t EbmlMaster::UpdateSize(const ShouldWrite & writeFilter, bool bForc
     const std::uint64_t SizeToAdd = Element->ElementSize(writeFilter);
 #if !defined(NDEBUG)
     if (static_cast<std::int64_t>(SizeToAdd) == (0-1))
```

Comment:
You have missing this one. And then if they are equal we can use the shorter form (static_cast in this case).

Reply:
this is an int not a uint. It can be negative.

Reply:
Indeed, it should be handled in a separate commit.

BTW UpdateSize is supposed to return a `filepos_t`.